kubectl-pg/cmd/scale.go

```diff
@@ -72,15 +78,22 @@ func scale(numberOfInstances int32, clusterName string, namespace string) {
 	} else {
 		log.Fatalf("cannot scale to the provided instances as they don't adhere to MIN_INSTANCES: %v and MAX_INSTANCES: %v provided in configmap or operatorconfiguration",maxInstances,minInstances)
 	}
+
 	if numberOfInstances == 0 {
 		fmt.Printf("Scaling to zero leads to down time. please type %s/%s and hit Enter this serves to confirm the action\n", namespace, clusterName)
 		confirmAction(clusterName, namespace)
 	}
-	postgresql.Labels = map[string]string{"spilo-role": "replica"}
-	UpdatedPostgres, err := postgresConfig.Postgresqls(namespace).Update(postgresql)
+
+	postgresJson, err := json.Marshal(postgresql)
+	if err != nil {
+		log.Fatal(err,"unable to parse json")
+	}
+
+	UpdatedPostgres, err := postgresConfig.Postgresqls(namespace).Patch(postgresql.Name,types.MergePatchType, []byte(postgresJson),"")
```

Comment:
Let's think about what happens here. To fix the original issue, we need to avoid explicitly setting default values that are present in the manifest returned by `Get` (that is, in the `postgresql` variable). Simply turning this entire variable into the patch (`postgresJson` variable) does not work regardless if one uses `Update` or `Patch` call to alter the manifest because the default values are still present in the patch content. The `Patch` call, however, enables us to change only the part of the manifest (number of instances in this case).So the idea   is do sth like that (pseudocode):

```golang
patchJson := "{number of instances : 3}"
patch, err := json.Marshal(patchJson)
if err != nil {
		log.Fatal(err,"unable to parse json")
	}

UpdatedPostgres, err := postgresConfig.Postgresqls(namespace).Patch(postgresql.Name,types.MergePatchType, []byte(patch),"")
```

Reply:
changed accordingly.